synchronization/src/cyface/java/de/cyface/synchronization/CyfaceAuthenticator.kt

```diff
@@ -0,0 +1,317 @@
+ * (at your option) any later version.
+ *
+ * The Cyface SDK for Android is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface SDK for Android. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.synchronization
+
+import android.accounts.AbstractAccountAuthenticator
+import android.accounts.Account
+import android.accounts.AccountAuthenticatorResponse
+import android.accounts.AccountManager
+import android.accounts.NetworkErrorException
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import androidx.annotation.MainThread
+import androidx.annotation.WorkerThread
+import de.cyface.uploader.Authenticator
+import net.openid.appauth.AppAuthConfiguration
+import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationService
+import net.openid.appauth.ClientAuthentication
+import net.openid.appauth.TokenRequest
+import net.openid.appauth.TokenResponse
+
+/**
+ * The CyfaceAuthenticator is called by the [AccountManager] to fulfill all account relevant
+ * tasks such as getting stored auth-tokens, opening the login activity and handling user authentication
+ * against the Cyface server.
+ *
+ * **ATTENTION:** The [.getAuthToken] method is only
+ * called by the system if no token is cached. As our logic to invalidate token currently is in this method, we call it
+ * directly where we need a fresh token.
+ *
+ * @author Klemens Muthmann
+ * @author Armin Schnabel
+ * @version 5.0.0
+ * @since 2.0.0
+ */
+class CyfaceAuthenticator(private val context: Context, private val authenticator: Authenticator) :
+    AbstractAccountAuthenticator(context) {
+
+    /**
+     * The service used for authorization.
+     */
+    private var mAuthService: AuthorizationService
+
+    /**
+     * The authorization state.
+     */
+    private var mStateManager: AuthStateManager
+
+    /**
+     * The configuration of the OAuth 2 endpoint to authorize against.
+     */
+    private var mConfiguration: Configuration
+
+    init {
+        // Authorization
+        mStateManager = AuthStateManager.getInstance(context)
+        //mExecutor = Executors.newSingleThreadExecutor()
+        mConfiguration = Configuration.getInstance(context)
+        val config = Configuration.getInstance(context)
+        //if (config.hasConfigurationChanged()) {
+            // This happens when starting the app after a fresh installation
+            //throw IllegalArgumentException("config changed (CyfaceAuthenticator)")
+            /*show("Authentifizierung ist abgelaufen")
+            Handler().postDelayed({signOut()}, 2000)*/
+            //return
+        //}
```

Comment:
FIXME: Replace `Configuration` with simple SharedPreferences like before
- but logout the user when the configuration changes (especially the URL)